protocol/database-scripting/README.md

```diff
@@ -0,0 +1,102 @@
+Database Scripting
+==============
+
+A protocol for running console scripts on the Healthify app's production database.
+
+Development
+-----------
+
+Develop the scripting story similarly to any other story. In this case, this would
+entail running your script at the rails console of your local server.
+
+While some scripts only lend themselves to being performed via pure SQL, where it
+is possible, you should leverage ActiveRecord to perform our INSERT/UPDATE/DELETE
+operations so that we leverage ActiveRecord validations and callbacks.
+
+While developing the script, it is critical to plan our remediation strategy if
+something does go awry.
+
+One useful approach to consider is to wrap the INSERT/UPDATE/DELETE commands within
+a transaction, and then to assert on certain states of the database after the script
+is complete, rolling back all changes if the assertions fail.
+
+Further, it's important to be able to execute a complete reversal of your changes
+in the event that there are unexpected failures. Ideally, we would write a reversal
+script to immediately reverse our changes if necessary.
+
+For INSERT operations, we would need to retain data about what records ended up
+being INSERT-ed by our script, and then the reversal script would delete those
+records.
+
+For UPDATE/DELETE operations, we would want to dump the data of all the records
+we will be interacting with into a Temp Table (or Tables). Our reversal script would then
+perform a COPY of the data from the Temp Table into the original table (and a deletion
+of updated records in the case of an Update).
+
+Code Review
+-----------
+
+Typically, console scripts are not code that we want to add to the Healthify app's
+code repo. As a result, to solicit Code Review you can put your code in a Github
+[gist](https://help.github.com/articles/about-gists/) and share the gist's link
+with your teammates to get Code Review. Also, be sure to include a link to the gist
+in the description of the corresponding Pivotal Story.
+
+Running database scripts poses a higher risk than deploying a normal feature. Hence,
+it's important to have a higher bar of Code Review than for normal stories. Make sure
+you solicit Code Review from at least two independent reviewers.
```

Comment:
> Thoughts on adding more CR vs. more QA.

All great points. I agree it's hard to know when to stop QAing, especially because it's possible (and probable) that our QA does not closely mimic a normal user's behavior (so it's feasible we totally miss a regular workflow that leads to identifying a bug) and also because our staging data is not regularly synced with production data, so the chances of us catching production data edge-cases decreases every day the two datasets stay diverged. I'm not sure what to do about the former, beyond perhaps we should require that more people do independent and/or paired QA. To mitigate the latter, I want to suggest that we more frequently backup our staging database with a sanitized production backup, or at least do so before testing a database script on staging. Thoughts?